crypto/rand/sysrand.c

```diff
@@ -88,21 +92,16 @@ int GFp_sysrand_chunk(void *out, size_t requested) {
 #endif
 #endif
 
-
-/* http://man7.org/linux/man-pages/man2/getrandom.2.html: "Calling
- * getrandom() to read /dev/urandom for small values (<= 256) of buflen is
- * the preferred mode of usage." */
-const size_t GFp_sysrand_chunk_max_len = 256;
-
 int GFp_sysrand_chunk(void *out, size_t requested) {
-  assert(requested <= GFp_sysrand_chunk_max_len);
-  if (syscall(SYS_getrandom, out, requested, 0u) < 0) {
-    if (errno == ENOSYS) {
-      return -1;
+  int r = syscall(SYS_getrandom, out, requested, 0u);
```

Comment:
Also, according to http://man7.org/linux/man-pages/man2/getrandom.2.html, the flags argument type is `unsigned int`. I suggest that we write this:
```unsigned int NO_FLAGS = 0;
    syscall(..., NO_FLAGS);
```
This way, we are sure there's no funny business going on.